subprojects/file-watching/src/integTest/groovy/org/gradle/internal/watch/WatchedDirectoriesFileSystemWatchingIntegrationTest.groovy

```diff
@@ -323,6 +376,7 @@ class WatchedDirectoriesFileSystemWatchingIntegrationTest extends AbstractFileSy
         then:
         projectDir.file('build/foo-1.0.jar').assertIsCopyOf(m2Module.artifactFile)
         assertWatchedHierarchies([projectDir])
+        assertNoContentRemovedUnderNewWatchableHierarchies()
```

Comment:
I moved the assertion to the other checks, so we don't need to sprinkle it around. Doing it in cleanup doesn't really make sense, since we need to do this after each build. The assertion also only helps when the build did run with `--info`.